Puzzling/Puzzling/Data/Model/TeamPuzzleStatusModel.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  TeamPuzzleStatusModel.swift
+//  Puzzling
+//
+//  Created by KYUBO A. SHIM on 2023/07/16.
+//
+
+import Foundation
+
+protocol ModelProtocol {}
+
+protocol TeamModelProtocol: AnyObject {
+    var status: TeamPuzzleStatusData { get set }
+    var puzzleCount: Int { get set }
+    var puzzleAssetNumber: Int { get set }
+}
+
+class TeamPuzzleStatusModel: TeamModelProtocol, ModelProtocol {
```

Comment:
`TeamModelProtocol`이나 `IndivisualModelProtocol` 프로토콜이 `ModelProtocol`의 하위 객체로 볼 수 있지 않나요? 
정확히 어떤 목적으로 만든지는 파악이 안되지만, 각 StatusModel에서 특정 모델 프로토콜을 채택하는 방식이라면,

```suggestion
protocol ModelProtocol {}

protocol TeamModelProtocol: ModelProtocol {
    var status: TeamPuzzleStatusData { get set }
    var puzzleCount: Int { get set }
    var puzzleAssetNumber: Int { get set }
}

class TeamPuzzleStatusModel: TeamModelProtocol {
```
로 계층을 가진 구조로 바꿔도 되지 않을까 싶어요!

Reply:
감사합니다! 현재 올라갈 #46 네트워크를 붙이면서 바뀌긴 했습니다.
해당 모델에 대해서 ModelProtocol 을 채택한건, 공통 컴포넌트에서 비슷하지만 다른 모델의 데이터를 썼어야 했는데
그 사용하는 타입을 하나로 지정하고 싶어서, ModelProtocol 을 만들어서 진행했었습니다!